axlearn/cloud/gcp/job.py

```diff
@@ -425,6 +428,15 @@ def _build_container(self) -> Nested[Any]:
         if cfg.enable_tpu_ici_resiliency is not None:
             env_vars["ENABLE_ICI_RESILIENCY"] = str(cfg.enable_tpu_ici_resiliency).lower()
 
+        resources = {"limits": {"google.com/tpu": system.chips_per_vm}}
+        # Set request memory by host machine type.
+        machine_memory_gb = GCE_MACHINE_TYPE_TO_REQUEST_MEMORY_CHARACTERISTICS.get(
+            system.gce_machine_type, None
+        )
+        if machine_memory_gb is not None:
+            resources["limits"]["memory"] = f"{machine_memory_gb}G"
+            resources["requests"] = {"memory": f"{round(machine_memory_gb * 0.8, 2)}G"}
```

Comment:
Does it mean that we are reserving 20% of memory for system software?

Reply:
Yes.